src/isZodTypeEqual.tsx

```diff
@@ -12,16 +12,20 @@ import {
 import { RTFSupportedZodTypes } from "./supportedZodTypes";
 import { unwrap } from "./unwrap";
 
-export function isZodTypeEqual(
+export function isZodTypeEqualImpl(
   _a: RTFSupportedZodTypes,
-  _b: RTFSupportedZodTypes
+  _b: RTFSupportedZodTypes,
+  visitedTypes: Set<RTFSupportedZodTypes>
 ) {
   // Recursively check objects
   // if typeNames are equal Unwrap Appropriate Types:
   // optional
 
   let { type: a, _rtf_id: idA } = unwrap(_a);
   let { type: b, _rtf_id: idB } = unwrap(_b);
+  if (visitedTypes.has(a) && visitedTypes.has(b)) return true;
```

Comment:
this allows for equality checking even in circular objects. if both a and b have been seen that means a call higher up in the stack is responsible for checking their equality. if every other property on the schema is equal for that call ten they are equal so returning true here defers that check to the call higher up. (and prevents an infinite loop ofc :P )